### PR TITLE
OD seg-faults when annotation labels contains "None" in them.

### PR DESCRIPTION
**Pull request by jakesabathia2**
fix  https://github.com/apple/turicreate/issues/2827
fix https://github.com/apple/turicreate/issues/2828

### CODE REVIEW

File: src/python/turicreate/test/test_object_detector.py [hunk 1]
```diff
@@ -313,6 +329,21 @@ def test_evaluate_missing_annotations(self):
             del sf[self.annotations]
             self.model.evaluate(sf.head())
 
+    def test_evaluate_with_missing_annotations_label(self):
+
+        def create_missing_annotations_label(x):
+            for y in x:
+                y['label'] = None
+            return x
```
Comment (Jarvi-Izana) [inline src/python/turicreate/test/test_object_detector.py]: well, 2 newlines are recommended as the interval between function the other codes.

File: src/toolkits/object_detection/od_data_iterator.cpp [hunk 2]
```diff
@@ -103,6 +103,10 @@ std::vector<image_annotation> parse_annotations(
 
       if (key == "label") {
         // If the label is invalid (not in class_to_index_map) then ignore it.
+        if (kv.second.get_type() == flex_type_enum::UNDEFINED) {
+          log_and_throw("Annotation labels must have valid type.");
```
Comment (Jarvi-Izana) [inline src/toolkits/object_detection/od_data_iterator.cpp]: add more informational text like: please consider 'dropna()' to clean column <blah>. Also, Check #2804 to early error out user the missing value will halt the training.

File: src/python/turicreate/test/test_object_detector.py (hunk 1 above)
Comment (jakesabathia2) [reply src/python/turicreate/test/test_object_detector.py]: good

File: src/toolkits/object_detection/od_data_iterator.cpp (hunk 2 above)
Comment (jakesabathia2) [reply src/toolkits/object_detection/od_data_iterator.cpp]: good!

Comment (jakesabathia2) [reply src/toolkits/object_detection/od_data_iterator.cpp]: will not do the same thing in #2804 in `init_training` tho,
bc OD allows empty annotation (they ignore them and this is covered in a test case),
and the missing value for image is handled in the Python side.

File: src/toolkits/object_detection/od_data_iterator.cpp [hunk 3]
```diff
@@ -103,6 +103,10 @@ std::vector<image_annotation> parse_annotations(
 
       if (key == "label") {
         // If the label is invalid (not in class_to_index_map) then ignore it.
+        if (kv.second.get_type() == flex_type_enum::UNDEFINED) {
+          log_and_throw("Annotation label is missing.");
```
Comment (srikris) [inline src/toolkits/object_detection/od_data_iterator.cpp]: "All annotations must contain label information."

**Comment by srikris** [comment]
Passes internal gitlab.

**Pull request merged**